File: scala/src/main/scala/mesosphere/rendler/Results.scala
package mesosphere.rendler

import play.api.libs.json._

/**
  * The type of crawl results
  */
case class CrawlResult(taskId: String, url: String, links: Seq[String])

/**
  * The type of render results
  */
case class RenderResult(taskId: String, url: String, imageUrl: String)

/**
  * JSON Formatters for executor results
  */
trait ResultProtocol {
  implicit val crawlResultFormat = Json.format[CrawlResult]
  implicit val renderResultFormat = Json.format[RenderResult]
}

File: go/src/github.com/mesosphere/rendler/results.go
package rendler

// CrawlResult is the result of a crawl operation
type CrawlResult struct {
	TaskID string
	URL    string
	Links  []string
}

// RenderResult is the result of a render operation
type RenderResult struct {
	TaskID   string
	URL      string
	ImageURL string
}
